src/bitcoind/actions.rs

```diff
@@ -286,30 +294,34 @@ fn poll_bitcoind(revaultd: &mut RevaultD, bitcoind: &BitcoinD) -> Result<(), Bit
             vault_tx,
         )
         .map_err(|e| BitcoindError(format!("Database error: {}", e.to_string())))?;
-        revaultd.vaults.insert(outpoint, utxo);
+        revaultd.write().unwrap().vaults.insert(outpoint, utxo);
 
         // Mind the gap! https://www.youtube.com/watch?v=UOPyGKDQuRk
         // FIXME: of course, that's rudimentary
-        if derivation_index > revaultd.current_unused_index {
-            revaultd.current_unused_index += 1;
-            let next_addr =
-                bitcoind.addr_descriptor(&revaultd.last_deposit_address().to_string())?;
+        if derivation_index > revaultd.read().unwrap().current_unused_index {
+            revaultd.write().unwrap().current_unused_index += 1;
+            let next_addr = bitcoind
+                .addr_descriptor(&revaultd.write().unwrap().last_deposit_address().to_string())?;
```

Comment:
```suggestion
            let mut revaultd = revaultd.write().unwrap();
            revaultd.current_unused_index += 1;
            let next_addr = bitcoind
                .addr_descriptor(&revaultd.last_deposit_address().to_string())?;
```

Reply:
I think it is better to do revaultd.write().unwrap() only once in the fn

Reply:
> I think it is better to do revaultd.write().unwrap() only once in the fn

While i agree it's syntactically more elegant, it would technically block the Mutex for the entire duration of the function, potentially starving other threads. No big deal *right now* but i think it's cleaner to block the Mutex as little time as possible.

Reply:
fair enough